compliance_checker/cf/util.py

```diff
@@ -392,36 +392,6 @@ def is_vertical_coordinate(var_name, var):
     return satisfied
 
 
-def compare_unit_types(specified, reference):
```

Comment:
Looks like there are other functions in `compliance_checker/cfutil.py` that perform similar tasks with more granularity. While this is a public function, I doubt compliance-checker users call these functions directly.

@benjwadams are you OK removing this? If not, we should probably add some test coverage for it.